### PR TITLE
Cleanup release script and integrate docs

### PR DESCRIPTION
**Pull request by tlrobinson**
Refactors release script a bit and adds docs. Haven't had a chance to fully test this end to end since we haven't done a release yet. We could leave this PR open until we do the next release.

### CODE REVIEW

File: release.sh [hunk 1]
```diff
@@ -39,69 +42,139 @@ if [ -z ${AWS_DEFAULT_PROFILE+x} ]; then
...
+        git clone "git@github.com:metabase/metabase-buildpack.git" "$METABASE_BUILDPACK_DIR"
+    fi
+
+    cd "$METABASE_BUILDPACK_DIR"
+    git fetch
+    git checkout "$METABASE_BUILDPACK_BRANCH"
+    git pull
+}
+
+pull-docs() {
```
Comment (salsakran) [inline release.sh]: tiny nit, but this is more accurately "pull-website" rather than just docs

File: release.sh [hunk 2]
```diff
@@ -39,69 +42,139 @@ if [ -z ${AWS_DEFAULT_PROFILE+x} ]; then
...
+    git tag "$VERSION"
+    git push
+    git push --tags origin "$METABASE_BUILDPACK_BRANCH"
+}
+
+release-docs() {
+    cd "$METABASE_WEBSITE_DIR"
+
+    echo "release docs"
+    ./script/docs "v$VERSION"
```
Comment (salsakran) [inline release.sh]: this doesn't do "--latest", is that on purpose? If so, there would still need to be a manual step of marking that version the latest.

Comment (tlrobinson) [reply release.sh]: I was thinking `release.sh` would not do `--latest` but `publish-release.sh` would, but I forgot to actually implement that in `publish-release.sh`.

Maybe we should also rename `release.sh` to `prepare-release.sh` to make it clearer what the purpose is?

* `release.sh` (or `prepare-release.sh`) does the build and put all the assets in place.
* `publish-release.sh` update the Docker `latest` tag, and it should update the website with links to the latest jar and `docs/latest`, etc (probably also push the blog post, `version-info` Github release, maybe even tweet?)

**Comment by salsakran** [comment]
I like the idea of calling it prepare-release.sh a lot

**Pull request merged**